sdk/avalon_sdk/ethereum/contracts/proxy-model/WorkOrderReceiptRegistry.sol

```diff
@@ -0,0 +1,229 @@
+/* Copyright 2020 iExec Blockchain Tech
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+pragma solidity ^0.5.0;
+
+import './LibSet.bytes32.sol';
+
+contract WorkOrderReceiptRegistry
+{
+	using LibSet_bytes32 for LibSet_bytes32.set;
+
+	// Structures
+	struct WorkOrderReceipt
+	{
+		bool    active;
+		bytes32 workerServiceId;
+		bytes32 workerId;
+		bytes32 requesterId;
+		uint256 receiptCreateStatus;
+		bytes   workOrderRequestHash;
+		uint256 currentReceiptStatus;
+		mapping(bytes32 => uint256[]) updates;
+	}
+
+	struct WorkOrderReceiptUpdate
+	{
+		bytes32 updaterId;
+		uint256 updateType;
+		bytes   updateData;
+		bytes   updateSignature;
+		string  signatureRules;
+	}
+
+	// Constants
+	uint256 constant internal LASTUPDATE = 0xFFFFFFFF;
+
+	// Members
+	mapping(bytes32 => WorkOrderReceipt)                                                               internal m_workorders;
+	WorkOrderReceiptUpdate[]                                                                           internal m_workorderupdates;
+	// workerServiceId → workerId → requesterId → receiptCreateStatus → Set<workOrderId>
```

Comment:
Do no use the arrow character. Replace with **->**. Causes issues in parsing at times.  Fixed earlier - https://github.com/hyperledger/avalon/commit/470c9124a0f481f7b5648c94847dc4bdd651fcc9#diff-98650fb1e6bb937efadc46d851b2e21bR40.